### PR TITLE
Add sample for enroll auth acr flow

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
Adds sample showing usage of Authenticator Enrollment flow.

**Usage (Root/Org AS):**

```
~/okta/samples-java-spring/okta-hosted-login$ mvn -Dokta.oauth2.issuer=https://example.com    -Dokta.oauth2.clientId=test12345     -Dokta.oauth2.clientSecret=supersecret      -DacrValues=urn:okta:loa:2fa:any:ifpossible   -DenrollAmrValues="sms okta_verify"  -DenrollmentCallbackUri="/callback/enroll"
```

**App Log:**

```
2022-12-09 22:25:27.771  INFO 14850 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2022-12-09 22:25:27.791  INFO 14850 --- [           main] c.o.s.e.CodeFlowExampleApplication       : Started CodeFlowExampleApplication in 6.412 seconds (JVM running for 10.066)
2022-12-09 22:25:36.855  INFO 14850 --- [nio-8080-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2022-12-09 22:25:36.856  INFO 14850 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2022-12-09 22:25:36.868  INFO 14850 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 12 ms
2022-12-09 22:25:46.381  INFO 14850 --- [nio-8080-exec-6] c.o.s.e.CodeFlowExampleApplication       : Enrolling Authenticator for someone@company.com
2022-12-09 22:25:46.387  INFO 14850 --- [nio-8080-exec-6] com.okta.spring.example.Util             : The issuer URL: 'https://example.com' is an Okta root/org authorization server
2022-12-09 22:25:46.388  INFO 14850 --- [nio-8080-exec-6] c.o.s.e.CodeFlowExampleApplication       : Sending Authorize request to: https://example.com//oauth2/v1/authorize?response_type=none&client_id=test12345&acr_values=urn:okta:loa:2fa:any:ifpossible&enroll_amr_values=sms okta_verify&prompt=enroll_authenticator&state=state12345&max_age=0&redirect_uri=http://localhost:8080/callback/enroll
2022-12-09 22:25:47.179  INFO 14850 --- [nio-8080-exec-6] c.o.s.e.CodeFlowExampleApplication       : Authorization result HTTP status code: 200
2022-12-09 22:27:22.330  INFO 14850 --- [nio-8080-exec-8] c.o.s.e.CodeFlowExampleApplication       : Enroll callback received with query string: state=state12345
```

### CODE REVIEW

File: okta-hosted-login/pom.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
         <dependency>
             <groupId>org.thymeleaf.extras</groupId>
             <artifactId>thymeleaf-extras-springsecurity5</artifactId>
-            <version>3.1.0.RELEASE</version>
+            <version>3.0.4.RELEASE</version>
```
Comment (arvindkrishnakumar-okta) [inline okta-hosted-login/pom.xml:43]: Downgraded this due to runtime issue with the latest `3.1.0` version.

File: okta-hosted-login/src/main/java/com/okta/spring/example/Util.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+            if (tokenizedUri.length >= 2 &&
+                    "oauth2".equals(tokenizedUri[0]) &&
+                    Strings.hasText(tokenizedUri[1])) {
+                logger.info("The issuer URL: '{}' is an Okta custom authorization server", issuerUri);
+                return false;
+            }
+        }
+
+        logger.info("The issuer URL: '{}' is an Okta root/org authorization server", issuerUri);
```
Comment (bdemers) [inline okta-hosted-login/src/main/java/com/okta/spring/example/Util.java:28]:
```suggestion
        logger.debug("The issuer URL: '{}' is an Okta root/org authorization server", issuerUri);
```

File: okta-hosted-login/src/main/java/com/okta/spring/example/Util.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    static boolean isRootOrgIssuer(String issuerUri) throws MalformedURLException {
+        String uriPath = new URL(issuerUri).getPath();
+
+        if (Strings.hasText(uriPath)) {
+            String[] tokenizedUri = uriPath.substring(uriPath.indexOf("/")+1).split("/");
+
+            if (tokenizedUri.length >= 2 &&
+                    "oauth2".equals(tokenizedUri[0]) &&
+                    Strings.hasText(tokenizedUri[1])) {
+                logger.info("The issuer URL: '{}' is an Okta custom authorization server", issuerUri);
```
Comment (bdemers) [inline okta-hosted-login/src/main/java/com/okta/spring/example/Util.java:23]:
```suggestion
                logger.debug("The issuer URL: '{}' is an Okta custom authorization server", issuerUri);
```

**Comment by arvindkrishnakumar-okta** [comment]
We will not be supporting this flow per internal decision, hence dropping this PR.

**Pull request closed without merging**